### PR TITLE
Batch Export

### LINKED ISSUE #812: [FEATURE] Batch download items in web UI
**Describe the bug**
If a single item is selected in the web UI, the Download button is enabled and it can be downloaded. If two or more items are selected, the Download button is disabled and grayed out.

**To reproduce**
1. Select an item via its checkbox
viame_batch_dl_1

2. Select another item
viame_batch_dl_2

**Related data**
https://viame.kitware.com/#/folder/5e4c2817a0fc86aa0315b1a1

**Expected behavior**
Allow download when multiple items are selected.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Browser: reproduced on Brave 1.25.72 (Chromium: 91.0.4472.101) and Chrome 91.0.4472.101

### LINKED ISSUE #925: [FEATURE] Ability to download multiple folders of data at once or folder of folders
Downloaded folders would ideally also be compatible with VIAME project train structure which can accept data in any number of ways (examples below though we may be able to add other structures to train tool if difficult)

root folder
--sequence1_folder
----image1.png
----image2.png
----annotations.csv   (annotations in folder)
--sequence2_folder
----image1.png
----image2.png
--sequence2_folder.csv  (annotations outside of folder at root level same name)
--video1.avi
--video1.csv  (annotations for video, same name)
--great_folder
----video2.avi  (video in subfolder)
----video2.csv

### PR DESCRIPTION
**Pull request by BryonLewis**
fixes #812 
fixes #925

Updated 20220131:

- Removed the `GET dive_annotation/{id}/export` for an endpoint `GET dive_annotation/export` where an array of `folderIds` is passed in.  Endpoint will now load the multiple folders and pass them to an updated `export_datasets_zipstream`.  If there are multiple folders it names the output `batch_export.zip` if not it will use the single folder's name.
- updated `export_datasets_zipstream` to `export_datasets_zipstream` to support multiple girder folders.  It will then use the settings and zip up the multiple folders instead of a single folder.  Just a minor restructuring of the function to support multiple datasets.
- Client side `platform/web-girder/views/Export.vue`  will now take in an array of folderIds instead of a single one.
- Behavior for a single dataset is the same using a new Ref `singleDatasetId` for multiple datasets there is only the option to export everything.
- Updated all references to the web-girder/Export.vue to pass in an array instead of a single datasetId.
- Added some integration tests that will test the regular export zip and batch export.  I avoided the `testCharacters` user because the names of the exported files don't match-up properly.

### CODE REVIEW

**Comment by subdavis** [comment]
> duplicate all of our existing export options (configuration, annotation, media, everything) for multiple datasets Or only allow full export when selecting multiple datasets?

full export only, IMO

> If you select multiple datasets you have the same options as before except all data will be downloaded into a single zip file instead of just downloading configuration files or oannotations.

Yep

> Behavior for selecting a single dataset should remain the same as it was before

Yes

> Selecting a non-folder dataset will git the standard https://viame.kitware.com/api/v1/resource/download endpoint

Out of scope, IMO.  We don't support this now and shouldn't add it.   The thing it would download is almost certainly not what the user wants.

> New endpoint

Yeah, just change the params of the existing one to accept an array, good call.  I don't even think you need to rename it.

**Comment by BryonLewis** [comment]
I still believe that we need to have separate endpoints for exporting a single dataset vs multiple.  Mostly because I like the clarity of `GET /dive_dataset/{id}/export` and then having an additional one with less options `GET /dive_dataset/batch_export` which takes an array of folders but doesn't have the other options.

I think in implementation I would modify the `export_dataset_zipstream` to `export_datasets_zipstream` where instead of a single `girder folder` it takes an array of folders and adds them to a single zip file.

**Review by subdavis — CHANGES_REQUESTED** [review]
Here's another small edge case:

Batch export for clones where the source media is missing will fail in a confusing/unclear way.

Before, if you tried to export a dataset where the clone chain is broken it looked like this.

Screenshot from 2022-01-31 11-56-22

But bulk export will let you include a broken clone (expected).  I think the resulting ZIP needs to have a "failures.txt" file or the failed dataset's folder needs to have a placeholder file that says what happened  If you have a better idea, that's cool too.

File: client/platform/web-girder/views/Home.vue [hunk 1]
```diff
@@ -89,9 +89,6 @@ export default defineComponent({
       }
       return null;
     },
-    exportTargetId() {
-      return this.exportTarget?._id || null;
-    },
```
Comment (subdavis) [inline client/platform/web-girder/views/Home.vue:94]: this was the last reference of `exportTarget`, so that can be removed as well.

**Review by subdavis — APPROVED** [review]
Works perfectly.

One nit to remove some dead code.

**Review by subdavis — APPROVED** [review]

**Pull request merged**